packages/apps/client/src/model/UISegment.ts

```diff
@@ -63,6 +63,22 @@ export class UISegment {
 			.sort((a, b) => a.rank - b.rank)
 	}
 
+	get linesInOrderFiltered(): UILine[] {
+		return Array.from(this.lines.values())
+			.slice()
+			.filter(this.doesLineMatchFilter)
+			.sort((a, b) => a.rank - b.rank)
```

Comment:
Slight performance improvement when no filter is active:
```suggestion
		let lines = Array.from(this.lines.values())
		if (this.owner.filter) lines = lines.filter(this.doesLineMatchFilter)
		return  lines.sort((a, b) => a.rank - b.rank)
```